enhancements/20250107-cpu-memory-hotplug.md

```diff
@@ -0,0 +1,98 @@
+# CPU / Memory Hotplug
+
+## Summary
+
+This enhancement focues on leveraging KubeVirt's CPU and Memory hotplug feature to enable users to dynamically adjust the CPU and Memory resources of a VM.
+
+### Related Issues
+
+https://github.com/harvester/harvester/issues/5835
+
+## Motivation
+
+### Goals
+
+- Users can adjust the CPU and Memory resources of a VM dynamically.
+- Users can set maximum CPU and Memory resources for a VM. The dynamic adjustment of CPU and Memory resources will be limited by the maximum resources.
+
+### Non-goals
+
+N/A
+
+## Proposal
+
+The cpu / memory hotplug feature will be enabled by default. When creating a VM, users can set 4 fields: initial cpu / memory and maximum cpu / memory. The initial cpu / memory will be the initial resources allocated to the VM. The maximum cpu / memory will be the maximum resources that the VM can be adjusted to.
+
+When the VM is running, users can adjust the CPU and Memory resources of the VM. The adjustment will be limited by the maximum resources set when creating the VM. However, users cannot adjust the CPU and Memory resources to be less than the current resources, because it's not allowed in KubeVirt.
+
+### User Stories
+
+#### Story 1 - Vertical Scaling a VM without downtime
+
+There is a legacy workload which is not actively developed. The workload was not designed as a distributed system, so it's hard to scale out. Recently, there are some users using this application, and the workload is getting heavier. The current resources allocated to the VM are not enough to handle the workload. The administration wants to adjust the CPU and Memory resources of the VM without downtime.
+
+### User Experience In Detail
+
+1. Prepare an Image.
+
+2. On the VM creation dashboard, click the `Enable CPU / Memory Hotplug` button and the dashboard will show another two fields: Maximum CPU and Maximum Memory.
+
+3. Fulfill the VM creation form and click the `Create` button.
+
+4. After the VM is created, administrators can adjust the CPU and Memory resources by clicking `CPU / Memory Hotplug` on the VM details page.
```

Comment:
It will be really cool if the UI can present a slider with the initial and maximum values being the lower and upper bound :sunglasses:.